core/src/main/java/org/mule/runtime/core/internal/util/CompositeClassLoader.java

```diff
@@ -24,14 +24,19 @@
 import java.util.Enumeration;
 import java.util.List;
 
+import com.github.benmanes.caffeine.cache.Cache;
+
 /**
  * Classloader implementation that, given a set of classloaders, will first search for a resource/class in the first one. If it is
  * not found, it will try the next and so on until the resource/class is found or all classloaders have been tried.
  * <p>
  * For {@link #getResources(String)}, all the classloaders will be queried to get the union of all found resources.
  *
  * @since 1.0
+ * 
+ * @deprecated Use {@link MultiParentClassLoaderUtils} instead.
```

Comment:
Regarding the "*Utils": Is it intentional because you're recommending to use the utility methods for creation or it's a typo and you meant MultiParentClassLoader?

Reply:
it is intentional, since the Utils handlings the caching as the `from` method from `CompositeClassLoader` does